runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -83,9 +83,13 @@
         // to the correct version, reflecting any optimistic
         // updates.
         return proxy.getUnderlyingObject().access(
-                o -> (getWriteSetEntryList(proxy.getStreamID()) == null && // No updates
+                o -> (getWriteSetEntryList(proxy.getStreamID()).size() == 0 && // No updates
```

Comment:
Well, if you mark getWriteSetEntryList's (and MultiObjectSMREntry::getSMRUpdates) return as @NotNull, then there is no need for the null check.